src/services/authorization.rs

```diff
@@ -0,0 +1,69 @@
+use std::sync::Arc;
+
+use crate::databases::database::{Database, Error};
+use crate::errors::ServiceError;
+use crate::models::user::{UserAuthorization, UserId};
+
+pub struct AuthorizeService {
+    user_authorization_repository: Arc<DbUserAuthorizationRepository>,
+}
+
+impl AuthorizeService {
+    #[must_use]
+    pub fn new(user_authorization_repository: Arc<DbUserAuthorizationRepository>) -> Self {
+        Self {
+            user_authorization_repository,
+        }
+    }
+
+    /// Checks if the user has the right privileges to perform the requested action.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if unable to get the user
+    /// authorization data from the database or if the user
+    /// does not have the right privileges to perform the action.
+    pub async fn authorize_user(&self, user_id: UserId, admin_required: bool) -> Result<(), ServiceError> {
+        // Checks if the user exists in the database
+        let authorization_info = self
+            .user_authorization_repository
+            .get_user_authorization_from_id(&user_id)
+            .await?;
+
+        //If admin privilages are required, it checks if the user is an admin
+        if admin_required {
+            Self::authorize_admin_user(&authorization_info)
+        } else {
+            Ok(())
+        }
+    }
```

Comment:
Hi @mario-nt I think we should remove the `admin_required` parameter otherwise the handler has to know if the action must be performed by an admin. The main idea was removing the responsibility of knowing which role can do the task from the axum handlers.

This implementation extracts only part of the responsibility.

I have created a PR to show you what I had in mind for this issue:

https://github.com/torrust/torrust-index/pull/546

It's a lot of changes but only because I want to test the service and I wanted to create mocks for the user repository. But the service itself it's very simple. The main problem is all contexts are coupled with the authorization service for the time being. But we were discussing that. I think it's OK for now.